### DISCUSSION TITLE
Does not work on STM32F405 please make a firmware for this microcontroller

### DISCUSSION THREAD

**Discussion by Navigatoroleg** [revoxhere/duino-coin#1422] [General]
Does not work on STM32F405 please make a firmware for this microcontroller

**Comment by JK-Rolling** [comment]
STM32 code is at https://github.com/revoxhere/duino-coin/tree/master/Unofficial%20miners/STM32
try modify the Unofficial miners/STM32/src/main.cpp line:160 

before:
`Serial.print(String(result) + "," + String(elapsedT) + "," + clientId + "\n");`

after:
`Serial.print(String(result, 2) + "," + String(elapsedT, 2) + "," + clientId + "\n");`

let us know how it went. I do not have STM32 to test on.

**Reply by Olegppppp** [reply]
![12](https://user-images.githubusercontent.com/102281421/159898958-a06e0ffe-4e55-4a7c-bfa0-0f3d6046a3a1.gif)

**Reply by Navigatoroleg** [reply]
> STM32 code is at https://github.com/revoxhere/duino-coin/tree/master/Unofficial%20miners/STM32 try modify the Unofficial miners/STM32/src/main.cpp line:160
> 
> before: `Serial.print(String(result) + "," + String(elapsedT) + "," + clientId + "\n");`
> 
> after: `Serial.print(String(result, 2) + "," + String(elapsedT, 2) + "," + clientId + "\n");`
> 
> let us know how it went. I do not have STM32 to test on.

Olegppppp is my second account.

**Reply by Olegppppp** [reply]
how can this problem be solved with difficulty?

**Reply by Navigatoroleg** [reply]
> STM32 code is at https://github.com/revoxhere/duino-coin/tree/master/Unofficial%20miners/STM32 try modify the Unofficial miners/STM32/src/main.cpp line:160
> 
> before: `Serial.print(String(result) + "," + String(elapsedT) + "," + clientId + "\n");`
> 
> after: `Serial.print(String(result, 2) + "," + String(elapsedT, 2) + "," + clientId + "\n");`
> 
> let us know how it went. I do not have STM32 to test on.

![16481934865137085229163355719440](https://user-images.githubusercontent.com/33506633/160074909-530250c4-257c-4ab8-a19b-b8865f410f28.jpg)
After correction

**Reply by Navigatoroleg** [reply]
> STM32 code is at https://github.com/revoxhere/duino-coin/tree/master/Unofficial%20miners/STM32 try modify the Unofficial miners/STM32/src/main.cpp line:160
> 
> before: `Serial.print(String(result) + "," + String(elapsedT) + "," + clientId + "\n");`
> 
> after: `Serial.print(String(result, 2) + "," + String(elapsedT, 2) + "," + clientId + "\n");`
> 
> let us know how it went. I do not have STM32 to test on.

![16481935616788971146471874449777](https://user-images.githubusercontent.com/33506633/160075104-218e3d1f-3c4a-423a-83ee-15ec9cf9fd88.jpg)

**Comment by Olegppppp** [comment]
It has already gone better without an error, but writes that the problem is with complexity and is rejected 13:19:44 net0 Searching for the fastest node to connect to 13:19:44 net0 Retrieved mining node: bila-pool-2 13:19:46 net0 Is connected to the main server Duino-Coin (v3.0) 13:19:46 net0 MOTD: You are mining on Bilapool, have fun! 13:19:46 sys8 The AVR miner's stream is started using the DUCO-S1A (COM8) algorithm 13:19:47 avr8 Rejected (Incorrect difficulty) 0/1 (0%) * 00.0s * 28 kH/s @ diff 6 * ping 348ms 13:19:47 avr8 Rejected (Incorrect difficulty) 0/2 (0%) * 00.0s * 28 kH/s @ diff 16 * ping 201ms 13:19:47 avr8 Rejected (Incorrect difficulty) 0/3 (0%) * 00.1s * 28 kH/s @ di

**Comment by Olegppppp** [comment]
‖ During the last 234 seconds
                ‖ You've mined 0 shares (0.0 shares/s)
                ‖ Blocks found: 0
                ‖ With the hashrate of 29240 H/s
                ‖ In this time period, you've solved 6842218 hashes
                ‖ Total miner uptime: 9.0 minutes

 (Incorrect difficulty) 0/37 (0%) ∙ 00.0s ∙ 29 kH/s @ diff 16 ∙ ping 51ms

**Comment by Olegppppp** [comment]
stm32f405 very fast. need to make more difficulty
